core/src/main/java/com/graphhopper/routing/lm/LMRoutingAlgorithmFactory.java

```diff
@@ -0,0 +1,81 @@
+    private final LandmarkStorage lms;
+    private final Weighting prepareWeighting;
+    private final int defaultActiveLandmarks;
+    private final int numBaseNodes;
+
+    public LMRoutingAlgorithmFactory(LandmarkStorage lms, int defaultActiveLandmarks, int numBaseNodes) {
+        this.defaultAlgoFactory = new RoutingAlgorithmFactorySimple();
+        this.lms = lms;
+        this.defaultActiveLandmarks = defaultActiveLandmarks;
+        this.prepareWeighting = lms.getWeighting();
+        this.numBaseNodes = numBaseNodes;
+    }
+
+    @Override
+    public RoutingAlgorithm createAlgo(Graph g, AlgorithmOptions opts) {
+        RoutingAlgorithm algo = defaultAlgoFactory.createAlgo(g, opts);
+        return getPreparedRoutingAlgorithm(g, algo, opts);
+    }
+
+    private RoutingAlgorithm getPreparedRoutingAlgorithm(Graph qGraph, RoutingAlgorithm algo, AlgorithmOptions opts) {
+        int activeLM = Math.max(1, opts.getHints().getInt(Parameters.Landmark.ACTIVE_COUNT, defaultActiveLandmarks));
+        if (algo instanceof AStar) {
+            if (!lms.isInitialized())
+                throw new IllegalStateException("Initialize landmark storage before creating algorithms");
+
+            double epsilon = opts.getHints().getDouble(Parameters.Algorithms.AStar.EPSILON, 1);
+            AStar astar = (AStar) algo;
+            astar.setApproximation(new LMApproximator(qGraph, prepareWeighting, numBaseNodes, lms, activeLM, lms.getFactor(), false).
+                    setEpsilon(epsilon));
+            return algo;
+        } else if (algo instanceof AStarBidirection) {
+            if (!lms.isInitialized())
+                throw new IllegalStateException("Initialize landmark storage before creating algorithms");
+
+            double epsilon = opts.getHints().getDouble(Parameters.Algorithms.AStarBi.EPSILON, 1);
+            AStarBidirection astarbi = (AStarBidirection) algo;
+            astarbi.setApproximation(new LMApproximator(qGraph, prepareWeighting, numBaseNodes, lms, activeLM, lms.getFactor(), false).
+                    setEpsilon(epsilon));
+            return algo;
+        } else if (algo instanceof AlternativeRoute) {
+            if (!lms.isInitialized())
+                throw new IllegalStateException("Initialize landmark storage before creating algorithms");
+
+            double epsilon = opts.getHints().getDouble(Parameters.Algorithms.AStarBi.EPSILON, 1);
+            AlternativeRoute altRoute = (AlternativeRoute) algo;
+            altRoute.setApproximation(new LMApproximator(qGraph, prepareWeighting, numBaseNodes, lms, activeLM, lms.getFactor(), false).
+                    setEpsilon(epsilon));
+            // landmark algorithm follows good compromise between fast response and exploring 'interesting' paths so we
+            // can decrease this exploration factor further (1->dijkstra, 0.8->bidir. A*)
+            altRoute.setMaxExplorationFactor(0.6);
+        }
+        return algo;
```

Comment:
Are there any cases besides `AStar`, `AStarBidirection` and `AlternativeRoute` (especially since we removed the CH+LM combination for now?). If not this code could be simplified.

Reply:
I don't think there are more possibilities.